e2e/test/iothub/twin/TwinE2ETests.cs

```diff
@@ -379,7 +403,46 @@ public static async Task RegistryManagerUpdateDesiredPropertyAsync(string device
             await registryManager.CloseAsync().ConfigureAwait(false);
         }
 
-        private async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync(Client.TransportType transport, Func<DeviceClient, string, object, MsTestLogger, Task<Task>> setTwinPropertyUpdateCallbackAsync, object propValue)
+        private async Task Twin_ServiceSetsDesiredPropertyAndDeviceUnsubscribes(Client.TransportType transport, object propValue)
+        {
+            var propName = Guid.NewGuid().ToString();
+
+            Logger.Trace($"{nameof(Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync)}: name={propName}, value={propValue}");
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+            
+            // Set a callback
+            await deviceClient.
+                SetDesiredPropertyUpdateCallbackAsync(
+                    (patch, context) =>
+                    {
+                        Logger.Trace($"{nameof(SetTwinPropertyUpdateCallbackHandlerAsync)}: DesiredProperty: {patch}, {context}");
+                        
+                        // After unsubscribing it should never reach here
+                        Assert.IsNull(patch);
+
+                        return Task.FromResult<bool>(true);
+                    },
+                    null)
+                .ConfigureAwait(false);
+
+            // Unsubscribe
+            await deviceClient
+                .SetDesiredPropertyUpdateCallbackAsync(null, null)
+                .ConfigureAwait(false);
+
+            await RegistryManagerUpdateDesiredPropertyAsync(testDevice.Id, propName, propValue)
+                .ConfigureAwait(false);
+
+            await deviceClient
+                .SetDesiredPropertyUpdateCallbackAsync(null, null)
+                .ConfigureAwait(false);
+
+            await deviceClient.CloseAsync().ConfigureAwait(false);
```

Comment:
same comment here about the flow -> subscribe -> test that it is received -> unsubscribe -> test not received.

Also, why is
```csharp
await deviceClient
                .SetDesiredPropertyUpdateCallbackAsync(null, null)
                .ConfigureAwait(false);
```
called twice (L431 and L438)?

Reply:
Good catch. I'll remove duplicate code.